src/main/scala/com/wavesplatform/state2/Diff.scala

```diff
@@ -54,8 +54,12 @@ case class Diff(transactions: Map[ByteStr, (Int, Transaction, Set[Address])],
       m.combine(Map(acc -> set))
     }
     groupedByAcc
```

Comment:
To me, this looks like the only possible cause of `StackOverflowError`. I see no point in changing other occurrences, because `mapValues` result is never cached.